feat(new_homepage): introduce corpus search UI with advanced filters

elisa-a-v opened the pull request:
## Fixes

Relates to #5206 but doesn't fully resolve it. Pending tasks described below.

## Summary

This PR introduces the search UI in the homepage using the same components from the header search bar.

To achieve this, this PR also:
- Introduces a new `CorpusSearchForm` class and adds it to all new templates.
- Refactors and extends the Corpus Search component.
- Extends the Dialog component by introducing a sub-component for a trigger button.

**Minor improvements:**
- Added custom style for checkboxes which is automatically added to any input with `type="checkbox"`.
- Introduced a new utility `focus-container` that provides a focus style to an element when a child input is focused.

**What's still pending:**
- #6000
- #6002
- #6043

## About the changes
#### New `CorpusSearchForm` class

The new form class is heavily based on the existing `SearchForm`, but only includes the fields needed so far, and changes styles to use tailwind classes and match new design.

We use the `IncrementalNewTemplateMiddleware` to introduce the `search_form` with an instance of the new `CorpusSearchForm` to the context in all new templates. This is required because we now support searching from every single page, so we need the form everywhere.

Before adding the new `CorpusSearchForm` instance, we check the existence of the new template, and only modify the context when it does exist, otherwise we return the response as-is. This is important to avoid breaking some old views like the Case Law advanced search.

#### Corpus Search component updates

1. **Refactor mobile version.**
   Mobile dialog is more easily reused with a customizable trigger button, so we use the same component in the header (with the magnifier icon trigger button), and in the homepage (with the "Filters" trigger button)

2. **Introduce fully responsive advanced filters with scoped fieldsets.**
   Fieldsets are disabled unless the scope is active. This prevents entered values in one scope from also being sent when submitting the form in a different scope.

3. **Introduce new scope selector variant.**
   By default the scope selector remains a simple dropdown (soon to me enhanced with new design), and we now also support a tabbed variant which is the one used in the desktop version of the new homepage.

## Deployment

**This PR should:**
- [ ] `skip-deploy`
- [ ] `skip-web-deploy`
- [x] `skip-celery-deploy`
- [x] `skip-cronjob-deploy`
- [x] `skip-daemon-deploy`

## Screenshots

<details>
<summary>Mobile homepage</summary>

<img width="400" alt="Screenshot from 2025-07-24 14-30-48" src="https://github.com/user-attachments/assets/e15f56e5-3302-41a6-be6c-fe0ddb69d80b" />

</details>

### Advanced filters

#### Case Law

<details>
<summary>Mobile</summary>

<img width="400" alt="Screenshot from 2025-07-24 13-50-10" src="https://github.com/user-attachments/assets/29209517-0e65-47bd-a410-03140c9857da" />

</details>

<details>
<summary>Desktop</summary>

<img width="1000" alt="image" src="https://github.com/user-attachments/assets/e134ecd5-4413-4126-b8e8-8a0099d5d762" />

</details>

#### RECAP Archive

<details>
<summary>Mobile</summary>

<img width="400" alt="Screenshot from 2025-07-24 13-50-18" src="https://github.com/user-attachments/assets/47283fd5-e401-4f2e-87b9-3a8a00a49d04" />

</details>

<details>
<summary>Desktop</summary>

<img width="1000" alt="image" src="https://github.com/user-attachments/assets/33b7e067-7ceb-4311-bd4e-49dfe3ce34c3" />

</details>

#### Oral Arguments

<details>
<summary>Mobile</summary>

<img width="400" alt="image" src="https://github.com/user-attachments/assets/40572dff-2640-4f3a-83ef-eee46060e8d7" />

</details>
<details>
<summary>Desktop</summary>

<img width="1000" alt="image" src="https://github.com/user-attachments/assets/e79b91e5-cf1f-471a-9d81-2be16f1d57dd" />

</details>

#### Judges

<details>
<summary>Mobile</summary>

<img width="400" alt="Screenshot from 2025-07-24 13-59-01" src="https://github.com/user-attachments/assets/a08ce16c-32aa-401f-ad3f-1991018326e2" />

</details>

<details>
<summary>Desktop</summary>

<img width="1000" alt="image" src="https://github.com/user-attachments/assets/f2e184b3-0733-4b14-965a-75267158e413" />

</details>

## Demos

<details>
<summary>Mobile homepage</summary>

[Screencast from 07-24-2025 03:16:56 PM.webm](https://github.com/user-attachments/assets/fbca01f1-6576-4839-8d67-0122749f5081)

</details>

<details>
<summary>Accessing advanced search from header</summary>

[Screencast from 07-24-2025 04:09:44 PM.webm](https://github.com/user-attachments/assets/8bf85f09-d1a4-48f3-bb87-6ed21ab0d720)

</details>

<details>
<summary>Responsive layout</summary>

[Screencast from 07-24-2025 04:21:18 PM.webm](https://github.com/user-attachments/assets/e7c4018a-b152-4709-a265-16064c5041f3)

</details>

ERosendo commented on cl/search/templates/cotton/corpus_search/scope.html:
```diff
@@ -3,9 +3,54 @@
+    x-on:keydown.end.prevent="focusLast"
+  >
+    <template x-for="scope in searchScopes">
+      <button
+        x-on:click="selectScope"
+        x-bind:data-scope="scope.label"
+        x-bind:id="scope.label"
+        x-bind:aria-selected="isActiveScope"
+        type="button"
+        role="tab"
+        class="bg-white px-8 min-w-41 h-[58px] text-sm font-normal rounded-t-2xl text-greyscale-600 flex justify-center items-center"
+      >
+        <span class="flex flex-col items-start">
+          <span
+            x-bind:data-scope="scope.label"
+            x-bind:class="scopeTabClass"
+            x-text="scope.label"
+            class=""
+          ></span>
+          <span
+            x-show="isActiveScope"
+            x-bind:data-scope="scope.label"
+            x-text="scope.shortDescription"
+            class="text-xs text-greyscale-500"
+          ></span>
+        </span>
+      </button>
```
I noticed that our tabs currently use the same color for both the **selected** and **unselected** states. At first, I thought it might just be my impression, but I double-checked the Figma mockup and confirmed it that the inactive tab should use a slightly warmer tone. Here's a video for reference:



https://github.com/user-attachments/assets/a0f612c0-323e-4cba-a6a9-8591e2ae9ec6

ERosendo commented on cl/search/templates/cotton/corpus_search/filters/judges.html:
```diff
@@ -0,0 +1,76 @@
+      </div>
+    </div>
+    <div class="flex flex-col justify-start gap-1.5 [&_label]:text-sm [&_label]:font-medium [&_label]:text-greyscale-600">
+      <div
+        class="text-ellipsis overflow-hidden text-nowrap"
+        title="{{ search_form.born_before.label }}"
+      >
+        {{ search_form.born_before.label_tag }}
+      </div>
+      <div class="focus-container input-text flex justify-stretch items-stretch gap-2">
+        {% svg "calendar" class="w-5 h-5" %}
+        {{ search_form.born_before }}
+      </div>
+    </div>
```
Is there a specific reason we’re wrapping these date fields in a `<div>`? On extra-large screens, I noticed this causes a slight increase in the spacing between the search bar and the first row of filters. After inspecting the HTML, I saw that the `labels` inside the `<div>` are slightly taller than regular labels.

Here's a video showing the issue:

https://github.com/user-attachments/assets/06db1f15-8d8e-43d0-ab14-fe16d7e490ad


As shown in the video, a standard label has a height of 20px, while the label wrapped in a `<div>` ends up with 24px. If the intent behind the `<div>` is just to apply a class to the label, I believe we can simplify it like this:

```html
      <label for="{{ search_form.born_after.id_for_label }}" class="text-ellipsis overflow-hidden text-nowrap">
        {{ search_form.born_after.label }}
      </label>
      <div class="focus-container input-text flex justify-stretch items-stretch gap-2">
        {% svg "calendar" class="w-5 h-5" %}
        {{ search_form.born_after }}
      </div>
```

elisa-a-v replied:
Nah those are probably just a leftover from various layout experiments, no need for them so I removed them 🫡 good catch

ERosendo reviewed (changes requested):
@elisa-a-v thanks for this PR! The new filters component looks very close to the mockups, and tab navigation works as expected. In addition to my previous comments, I have two more notes:
 
----

1. In the Figma mockups, there's a component next to the scope selectors that appears to be an anchor link. I'm attaching a screenshot highlighting the element I'm referring to.

    This PR doesn’t introduce that highlighted element. I checked the document provided in issue #5206 and didn’t see any mention of it being removed. It’s also not listed among the pending tasks, so I’m not sure if it’s expected to be added in a follow-up PR or if it was intentionally dropped from the design, though it’s still visible in the mockups.

<img width="1156" height="54" alt="Screenshot 2025-07-26 at 10 34 30 PM" src="https://github.com/user-attachments/assets/22a1dbd6-f6ad-47ba-8976-0b2e8d01bc32" />

   


----

2. I noticed that the "Filters" trigger button is always present in the search bar within the header. On mobile, it appears below the keywords input. However, for larger breakpoints, the trigger should be integrated inside the keywords input itself.

   This PR is introducing the trigger button for the mobile version , but the trigger is missing on larger screen sizes. This task isn’t listed as pending, and there’s no clear indication that it was intentionally left out. I'm attaching a screenshot highlighting the missing element:

<img width="1325" height="323" alt="Screenshot 2025-07-26 at 10 52 05 PM" src="https://github.com/user-attachments/assets/34e85bc3-f212-42a8-9cbd-e676a264e906" />

 Just wanted to flag this in case the desktop version is still planned or was deferred.

elisa-a-v commented on cl/assets/templates/cotton/dialog/index.html:
Yeah I wouldn't worry about those to be honest. They were there before and we don't have any issues with the dialogs, it's just the CSP build being noisy.

Some built-in features aren't supported so some warnings here and there will be inevitable with our setup.

elisa-a-v commented on cl/assets/templates/cotton/header.html:
```diff
@@ -100,24 +101,13 @@
-            Search
-          </c-corpus-search.button>
-        </c-corpus-search.mobile>
+        <c-corpus-search.mobile.dialog :request="request" :search_form="search_form" only>
+          <c-slot name="trigger_button">
+            <c-dialog.trigger-button>
+              {% svg "magnifier" class="text-greyscale-400 w-5 h-5" %}
+            </c-dialog.trigger-button>
+          </c-slot>
+        </c-corpus-search.mobile.dialog>
```
This happened because the desktop header search bar doesn't have advanced filters yet, so the function that updates the fieldsets threw the error when trying to update a non-existent element. Now we simply return when no element is found.

ERosendo commented on cl/search/templates/cotton/corpus_search/mobile/dialog.html:
```diff
@@ -49,11 +49,18 @@
+            href="{% url "coverage" %}"
+            class="flex items-center justify-center text-sm text-greyscale-600 gap-1 hover:text-greyscale-500"
+          >
+            {% svg "circle_question_mark" class="text-greyscale-400 hover:text-greyscale-300 h-4 w-4" %}
```
It seems the SVG file is missing.

elisa-a-v commented:
Thanks again Eduardo for all your feedback!

Latest changes:
- Renamed the components library template to begin with `v2_`, following our standard and thus having the new search form injected in its context. This fixed an issue in which the advanced filters were not working in that page.
- We now close all dialogs when the window is resized past the mobile breakpoint to avoid weird UX (dialogs appearing out of nowhere just by resizing the window can be confusing), and to fix an accessibility issue in which the desktop form remained hidden to AT if the dialog was not closed when the window was resized.
- Refactored the `IncrementalNewTemplateMiddleware` to use early return and reduce indentation.

I've also opened the following issues for more pending work:
- #6085
- #6086

ERosendo reviewed (approved):
@elisa-a-v Thanks for the latest tweaks! I confirmed that the advanced filters are working on the components page, and the updated dialog behavior when resizing the window looks great. I also appreciate the refactor to use early returns, it's much cleaner.

:shipit:
